src/include/txn_inline.h

```diff
@@ -1194,6 +1194,19 @@ __wt_txn_id_check(WT_SESSION_IMPL *session)
     if (F_ISSET(txn, WT_TXN_HAS_ID))
         return (0);
 
+    /*
+     * Return error when the transactions with read committed or uncommitted isolation tries to
+     * perform any write operation. Skip metadata because metadata searches (like schema operations)
```

Comment:
The comments related to metadata may be enhanced to get more clarity. How about something like the following?
`Don't raise an error for an update on metadata as the search and updates on metadata happens in read-uncommitted and read-committed isolations compared to the other tables`.

Reply:
Updated comments.